### DISCUSSION TITLE
EJBCA CE Docker Image on MacOS

### DISCUSSION THREAD

**Discussion by watdaf** [Keyfactor/ejbca-ce#657] [General]
Attempting to install a Docker Image of the latest CE on MacOS (proof of concept) using these instructions: https://docs.keyfactor.com/ejbca/latest/tutorial-start-out-with-ejbca-docker-container

Having the following error when starting the Docker compose up -d

ejbca-database  | 2024-09-03 23:22:15+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:11.5.2+maria~ubu2404 started.
ejbca-database  | chown: changing ownership of '/var/lib/mysql/': Permission denied
ejbca-database  | 2024-09-03 23:24:24+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:11.5.2+maria~ubu2404 started.
ejbca-database  | chown: changing ownership of '/var/lib/mysql/': Permission denied

Here is my docker-compose.yml

networks:
  access-bridge:
    driver: bridge
  application-bridge:
    driver: bridge
services:
  ejbca-database:
    container_name: ejbca-database
    image: "library/mariadb:latest"
    networks:
      - application-bridge
    environment:
      - MYSQL_ROOT_PASSWORD=foo123
      - MYSQL_DATABASE=ejbca
      - MYSQL_USER=ejbca
      - MYSQL_PASSWORD=ejbca
    volumes:
      - ./datadbdir:/var/lib/mysql:rw
  ejbca-node1:
    platform: linux/amd64
    hostname: ejbca-node1
    container_name: ejbca
    image: registry.primekey.com/primekey/ejbca-ee:7.10.1
    depends_on:
      - ejbca-database
    networks:
      - access-bridge
      - application-bridge
    environment:
      - DATABASE_JDBC_URL=jdbc:mariadb://ejbca-database:3306/ejbca?characterEncoding=UTF-8&allowPublicKeyRetrieval=true
      - LOG_LEVEL_APP=INFO
      - LOG_LEVEL_SERVER=INFO
      - TLS_SETUP_ENABLED=simple
    ports:
      - "80:8080"
      - "443:8443"

**Comment by primetomas** [comment]
Two things. 
The error message would indicate that there is an issue with your volume for MariaDB.
volumes:
- ./datadbdir:/var/lib/mysql:rw

As the error is MariaDB related and not EJBCA:
chown: changing ownership of '/var/lib/mysql/': Permission denied

Second, your image for EJBCA is a very old EJBCA Enterprise, and no EJBCA Community as you say.
image: registry.primekey.com/primekey/ejbca-ee:7.10.1

**Reply by watdaf** [reply]
I have tried both old and recent versions of CE and EE with no difference in the results with regard to the DB issue.

**Reply by primetomas** [reply]
That is because it's not related to EJBCA. But to the MariaDB container. The message comes from the MariDB container, not from the EJBCA container. And the error is not related to EJBCA, but to the setup/start/configuration of the MariaDB.
